sycl/include/CL/sycl/INTEL/esimd/esimd_view.hpp

```diff
@@ -101,7 +101,7 @@ template <typename BaseTy, typename RegionTy> class simd_view {
 
   template <int Dim = 0> constexpr uint16_t getOffset() const {
     static_assert(Dim <= is2D(), "region is not two-dimensional");
```

Comment:
Your fix looks good.

Something is really wrong with the lines 103, 98, 93: (int <= bool).
Looking at the assert message I would think that the condition should be something like this:
```suggestion
    static_assert(Dim >= 0 && Dim < 2 && is2D(), "region is not two-dimensional");
```

Reply:
I agree it deserves refactoring.